src/unstract/sdk/file_storage/fs_impl.py

```diff
@@ -0,0 +1,180 @@
+from typing import Union
+
+import fsspec
+from fsspec import AbstractFileSystem
+
+from unstract.sdk.exceptions import FileOperationError
+from unstract.sdk.file_storage.constants import Common, FileSeekPosition
+from unstract.sdk.file_storage.fs_interface import FileStorageInterface
+from unstract.sdk.file_storage.helper import FileStorageHelper
+
+
+class FileStorage(FileStorageInterface):
+    # This class integrates fsspec library for file operations
+
+    fs: fsspec  # fsspec file system handle
+
+    def __init__(self, type, credentials: dict() = {}):
+        self.fs = FileStorageHelper.file_storage_init(
+            type=type, credentials=credentials
+        )
+
+    def open(
+        self,
+        path: str,
+        mode: str,
+        encoding: str = Common.DEFAULT_ENCODING,
+    ) -> Union[AbstractFileSystem]:
+        """Opens the specified file and returns the actual file handle based on
+        the library in use to perform file operations.
+
+        Args:
+            path (str): Path to the file to be opened
+            mode (str): Mode in whicg the file is to be opened. Usual options
+                        include r, rb, w and wb
+            encoding (str): Encoding type like utf-8 or utf-16
+
+        Returns:
+            Union[AbstractFileSystem] : Actual file handle to the file that was opened
+        """
+        try:
+            return self.fs.open(path=path, mode=mode, encoding=encoding)
+        except Exception as e:
+            raise FileOperationError(str(e))
+
+    def read(
+        self,
+        file_handle: Union[AbstractFileSystem],
```

Comment:
@hari-kuriakose @gaya3-zipstack the methods where we are using file_handle. Am just thinking is there any real benefit to making this an input? Couldn't we make this an abstraction and make the file_handle an instance variable?